core/src/language/kotlin.rs

```diff
@@ -426,6 +446,36 @@ impl Kotlin {
         )
     }
 
+    fn write_private_element(
```

Comment:
this function only differs from `write_element` by a single line. Instead of duplicating the function could we instead add a new argument to write_element. e.g.

```rust
enum Privacy {
    Public,
    Private
}

fn write_element(
        &mut self,
        w: &mut dyn Write,
        f: &RustField,
        generic_types: &[String],
        requires_serial_name: bool,
        privacy: Privacy
    ) -> std::io::Result<()> {
        ...
        write!(
            w,
            match privacy {
                Privacy::Public => "\tval {}: {}{}",
                Privacy::Private => "\tprivate val {}: {}{}"
            },
            ...
}
```